VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowDeployProvider.cs

```diff
@@ -143,25 +150,31 @@ private async Task<string> GetOutputPathAsync(string filename)
             return outputPath;
         }
 
-        private static void MeadowConnection_DeviceMessageReceived(object sender, (string message, string source) e)
+        private static async void MeadowConnection_DeviceMessageReceived(object sender, (string message, string source) e)
         {
-            _ = outputLogger.ReportDeviceMessage(e.message);
+            await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await outputLogger.ReportDeviceMessage(e.message);
+            });
         }
 
         private static async void MeadowConnection_DeploymentProgress(object sender, (string fileName, long completed, long total) e)
         {
-            uint p = 0;
-
-            if (e.total != 0)
-            {
-                p = (uint)(e.completed * 100f / e.total);
-            }
-            else
+            await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
Double check here as well

Reply:
In this case it was already inside ResetProgressBar(). Nicely spotted.